crates/zksync/core/src/vm/runner.rs

```diff
@@ -535,6 +537,14 @@ fn inspect_inner<S: ReadStorage + Send>(
         .map(|b| bytecode_to_factory_dep(b.original.clone()))
         .collect();
     let modified_keys = storage.borrow().modified_storage_keys().clone();
+    for key in modified_keys.iter() {
+        let mut data = recorded_writes.lock().unwrap();
+        let address = key.0.address().to_address();
+        let slot = h256_to_u256(*key.1);
+        if let Some(map) = data.as_mut() {
+            map.entry(address).or_insert_with(Vec::new).push(slot.to_alloy());
+        }
+    }
```

Comment:
This isn't correct, the `modified_keys` is a map of `StorageKey => H256` where `H256` is the new value of the storage key. The `StorageKey` itself has the `(key.address(), key.key())` part that corresponds to `(address, slot)`.

```rust
for (k, v) in &modified_keys {
    println!("address={:?} slot={:?} value={:?}", k.address(), k.key(), v);
}
```